src/btree/bt_curnext.c

```diff
@@ -904,8 +904,16 @@ __wt_btcur_next(WT_CURSOR_BTREE *cbt, bool truncating)
         if (page != NULL &&
           (cbt->page_deleted_count > WT_BTREE_DELETE_THRESHOLD ||
             (newpage && cbt->page_deleted_count > 0))) {
-            WT_ERR(__wt_page_dirty_and_evict_soon(session, cbt->ref));
-            WT_STAT_CONN_INCR(session, cache_eviction_force_delete);
+            /* If checkpoint is happening on the btree, we can only evict clean content. */
```

Comment:
What problem does it cause to add this to the eviction queue when checkpoint is running except we will have a few eviction failures?

Reply:
We are triggering unnecessary work. We know we cannot evict dirty content if a checkpoint is happening, so let's not try at all.

Reply:
Do we see any adversary effect on the user (increased latency or something)? BTW, how about adding this to the normal eviction queue instead?

Reply:
Any dirty eviction gets failed whenever the checkpoint is running, so there is no point in trying.

Reply:
A test reproducing this was developed using control points. See https://jira.mongodb.org/browse/WT-13541 and PR https://github.com/wiredtiger/wiredtiger/pull/11102. This test has been merged to the control points project branch, not to develop.